File: apps/admin-v2/src/application/use-cases/auth/register.use-case.ts
import type { AuthRepository } from '@/domain/repositories/auth.repository';

export class RegisterUseCase {
  constructor(private repo: AuthRepository) {}

  execute(data: { name: string; email: string; password: string; businessName?: string; businessType?: string }) {
    return this.repo.register(data);
  }
}

File: apps/admin_mobile/lib/application/use_cases/auth/register_use_case.dart
import '../../../domain/repositories/auth_repository.dart';

class RegisterUseCase {
  final AuthRepository _repo;
  RegisterUseCase(this._repo);

  Future<LoginResult> call({
    required String name,
    required String email,
    required String password,
  }) =>
      _repo.register(name: name, email: email, password: password);
}
